extensions/wikia/ContentReview/ContentReviewImportJS.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace Wikia\ContentReview;
+
+class ImportJS {
+	const IMPORT_SCRIPTS_ENTRYPOINT = 'ImportJS';
+	const IMPORT_SCRIPTS_FUNCTION = 'importWikiaScriptPages';
+	const IMPORT_SCRIPTS_KEY = 'content-review-importjs';
+	const IMPORT_SCRIPTS_VERSION = '1.0';
+
+	/**
+	 * Get js script to load imports
+	 *
+	 * @return String
+	 */
+	public function getImportScripts() {
+		$importScripts = \WikiaDataAccess::cache(
+			$this->getImportJSMemcKey(),
+			2592000, // 30 days,
+			function() {
+				$importScripts = '';
+				$title = \Title::newFromText( self::IMPORT_SCRIPTS_ENTRYPOINT, NS_MEDIAWIKI );
+
+				if ( $title instanceof \Title && $title->exists() ) {
+					$revision = \Revision::newFromTitle( $title );
+					$scripts = explode( PHP_EOL, $revision->getRawText() );
+
+					$imports = $this->prepareImports( $scripts );
+					$importScripts = $this->createInlineScript( $imports );
+				}
+
+				return $importScripts;
+			}
+		);
+
+		return $importScripts;
+	}
+
+	/**
+	 * Check and filter provided scripts.
+	 *
+	 * Names should end with '.js' extension.
+	 * External scripts are allowed only from dev.wikia.com
+	 *
+	 * @param array $scripts
+	 * @return array
+	 */
+	public function prepareImports( Array $scripts ) {
+		$imports = [];
+
+		foreach ( $scripts as $key => $script ) {
+			$script = trim( $script );
+
+			if ( strtolower( substr( $script, -3 ) ) === Helper::JS_FILE_EXTENSION
```

Comment:
This should just be `substr( $script, -3 ) === Helper::JS_FILE_EXTENSION`. Changing to `strtolower` when checking allows page names like "MediaWiki:Foo.Js" or "MediaWiki:Foo.JS" through which are not valid JS pages. Only MW pages ending in lowercase `.js` are valid, so no need to transform the input.